### PR TITLE
fix(buttons): #FOR-595 block user from spamming buttons

### PR DESCRIPTION
**Pull request by c4role**
## Describe your changes
Added ng-disable to prevent certain buttons spamming and consequences

## Checklist tests
Here is the list of impacted's buttons : 
edit-form : saveAll
prop-form : saveGuard
toaster-forms-list : duplicateForm 
forms-list : doExportForms 
forms-list : doArchiveForms 
respond-question : next 
recap-question : doSend  
captcha : doSend 

## Issue ticket number and link
[FOR-595](https://jira.support-ent.fr/browse/FOR-595)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: formulaire/src/main/resources/public/template/containers/results-form.html [hunk 1]
```diff
@@ -63,7 +63,7 @@ <h1>
                     </option>
                 </select>
             </div>
-            <button class="cell" ng-click="vm.next()" ng-disabled="vm.last">
+            <button class="cell" ng-click="vm.next()" ng-disabled="vm.last || vm.isProcessing">
```
Comment (qperie) [inline formulaire/src/main/resources/public/template/containers/results-form.html:66]: Une dizaine de lignes au-dessus faudrait faire pareil pour le button previous j'pense

File: formulaire-public/src/main/resources/public/ts/controllers/captcha.ts [hunk 2]
```diff
@@ -65,6 +66,7 @@ export const captchaController = ng.controller('CaptchaController', ['$scope',
     vm.doSend = async () : Promise<void> => {
         let distributionData = await responseService.sendResponses(vm.formKey, vm.distributionKey, vm.responseCaptcha, vm.responses);
         let distribution = Mix.castAs(Question, distributionData);
+        vm.isProcessing = true;
```
Comment (qperie) [inline formulaire-public/src/main/resources/public/ts/controllers/captcha.ts:69]: Faut le mettre au début de la fonction parce que si tu spam clic il va call plusieurs fois le responseService.sendResponses()

```suggestion
        vm.isProcessing = true;
        let distributionData = await responseService.sendResponses(vm.formKey, vm.distributionKey, vm.responseCaptcha, vm.responses);
        let distribution = Mix.castAs(Question, distributionData);
```

File: formulaire/src/main/resources/public/ts/controllers/forms-list.ts [hunk 3]
```diff
@@ -405,6 +409,7 @@ export const formsListController = ng.controller('FormsListController', ['$scope
 
     vm.doExportForms = async () : Promise<void> => {
         vm.display.loading.export = true;
+        vm.isProcessing = true;
```
Comment (qperie) [inline formulaire/src/main/resources/public/ts/controllers/forms-list.ts:412]: Si tu veux t'éviter de surcharger le code, coté html tu peux juste ng-disabled selon la variable "vm.display.loading.export" et comme ça pas besoin de vm.isProcessiong dans cette fonction !

File: formulaire/src/main/resources/public/ts/controllers/respond-question.ts [hunk 4]
```diff
@@ -162,12 +163,14 @@ export const respondQuestionController = ng.controller('RespondQuestionControlle
 
     vm.next = async () : Promise<void> => {
```
Comment (qperie) [inline formulaire/src/main/resources/public/ts/controllers/respond-question.ts:164]: Je rejoins mon com coté html, y'a aussi le prev() à faire 👍

File: formulaire-public/src/main/resources/public/template/containers/respond-question.html [hunk 5]
```diff
@@ -50,7 +50,7 @@ <h1>
             <div class="buttons">
                 <button class="cell" ng-if="vm.formElement.position <= 1" ng-disabled="true"><i18n>formulaire.public.prev</i18n></button>
                 <button class="cell cancel" reset-guard="vm.prev()" ng-if="vm.formElement.position > 1"><i18n>formulaire.public.prev</i18n></button>
-                <button class="cell" reset-guard="vm.next()"><i18n>formulaire.public.next</i18n></button>
+                <button class="cell" reset-guard="vm.next()" ng-disabled="vm.isProcessing"><i18n>formulaire.public.next</i18n></button>
```
Comment (qperie) [inline formulaire-public/src/main/resources/public/template/containers/respond-question.html:53]: Je vois pas de modif coté controller ?

**Review by qperie — CHANGES_REQUESTED** [review]

**Review by alicedraillard — APPROVED** [review]

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**